WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -493,6 +530,10 @@ private extension StatsPeriodStore {
 
         // 'limit' in this context is used for the 'num' parameter for the 'file-downloads' endpoint.
         // 'num' relates to the "number of periods to include in the query".
+        if FeatureFlag.statsNewAppearance.enabled {
+            group.enter()
+            DDLogInfo("Stats Period: Enter group fetching file downloads.")
```

Comment:
This is more of a nitpick:
  
Since we have all the `.enter` calls inside a `fetchAsyncData(..)` I would have all the `.leave` calls inside this method as well. Otherwise, `.leave` calls get scattered across all the `receive...()` methods which can be called from other contexts. While looking at those `receive..()` methods in isolation, it's unclear what is the use of the `DispatchGroup` and if it's supposed to be notified in those other contexts as well.

Example:

```
      if FeatureFlag.statsNewAppearance.enabled {
          group.enter()
          DDLogInfo("Stats Period: Enter group fetching likes summary.")
      }
      let likesOperation = PeriodOperation(service: service, for: period, date: date, limit: 14) { [weak self] (likes: StatsLikesSummaryTimeIntervalData?, error: Error?) in
          if error != nil {
              DDLogError("Stats Period: Error fetching likes summary: \(String(describing: error?.localizedDescription))")
          }

          DDLogInfo("Stats Period: Finished fetching likes summary.")
          DispatchQueue.main.async {
              self?.receivedLikesSummary(likes, error)
          }
          
          if FeatureFlag.statsNewAppearance.enabled {
               DDLogInfo("Stats Period: Leave group fetching likes summary.")
               group. Leave()
          }
      }
```

Reply:
👋 I agree so I went ahead and put the `.leave()` calls within the context of the `.enter()` calls. I don't think this changes the functionality in any way.

Reply:
Thanks. I'll check and approve if the rest is good 👍